### ISSUE TITLE
[core] Object type is used instead of template type

### ISSUE THREAD

**Issue by bagipro** [skylot/jadx#1193]
Hi,

I've noticed a simple case when `java.lang.Object` type is used instead of template `T` type in class `androidx.recyclerview.widget.SortedList`:
```java
    public void recalculatePositionOfItemAt(int i) {
        throwIfInMutationOperation();
        java.lang.Object obj = get(i); // T should be used here
        removeItemAtIndex(i, false);
        int add = add(obj, false);
        if (i != add) {
            this.mCallback.onMoved(i, add);
        }
    }
```

APK: https://drive.google.com/file/d/1qC3tlWs9AtPBpyS6iU9kcSphLryfJOxi/view

**Comment by skylot** [comment]
@bagipro this issue is also fixed by commit fef3e21c70801c6d4d0ffb6df9a8943767a123f7 from #1192.

**Comment by bagipro** [comment]
@skylot 
Thanks!
